main/src/main/scala/sbt/internal/DefaultBackgroundJobService.scala

```diff
@@ -508,6 +508,15 @@ private[sbt] object DefaultBackgroundJobService {
     backgroundJobServices.values.forEach(_.shutdown())
     backgroundJobServices.clear()
   }
+
+  private[sbt] def stop(): Unit = {
```

Comment:
I think it's better to interpret Ctrl-C to be stop the foreground jobs. So instead of stopping everything, it should stop only the jobs started via `run`, or something like it?

Reply:
Currently only `run` and `runMain` use `BackgroundJobService` (through `BackgroundJobService.runInBackgroundWithLoader`), so it effectively stops only the jobs started via `run` or `runMain`.

`JobHandle` has a `spawningTask` attribute though. Shall investigate if we can use `spawningTask.key` to do filtering there to specifically filter for `run` and `runMain`.

Reply:
What about `bgRun`?

Reply:
Btw I do wonder if we should add a new boolean flag `isForeground` to 
https://github.com/sbt/sbt/blob/1ee3c4b142c3bb26c4091fbacaee8ca5999ed19a/main/src/main/scala/sbt/BackgroundJobService.scala#L99-L103

So we can just filter by `isForeground` to tell if the background job should be stopped when `ctrl+c` is pressed

Reply:
> bgRun

Oh right, I completely forgot about it...

Reply:
Yea. I like the idea. `isAutoCancel` maybe, since the job itself is background.